src/design-system/components/Text/Text.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { ElementRef, forwardRef, ReactNode, useMemo } from 'react';
+import { Text as NativeText } from 'react-native';
+import {
+  CustomColor,
+  useForegroundColor,
+} from '../../color/useForegroundColor';
+import { createLineHeightFixNode } from '../../typography/createLineHeightFixNode';
+import {
+  nodeHasEmoji,
+  nodeIsString,
+  renderStringWithEmoji,
+} from '../../typography/renderStringWithEmoji';
+import { textSizes, textWeights } from '../../typography/typography';
+
+const validColors = [
```

Comment:
I am thinking if we can try to make "config" separated somewhere? E.g. under `/config` dir. I can imagine that those files can be considered as an `engine` that should be by default not touchable. And `/config` should be adjusted (e.g. when we will decide to change colors of the rainbow.

Reply:
I too like the idea of having these somewhere that doesn't feel "normal" for us to be editing or changing these. not a block now imo

Reply:
Good idea. Moving this to `typography.ts` since that's where I've defined similar text scales.